cypress/support/commands.ts

```diff
@@ -40,3 +40,70 @@ Cypress.Commands.overwrite('tab', (originalFn, subject) => {
       log.end();
     });
 });
+
+declare global {
+  namespace Cypress {
+    interface Chainable<Subject> {}
+
+    interface Chainer<Subject> {
+      /**
+       * Asserts the element has an aria description matching the text
+       * @example
+       *    cy.get('body').should('have.ariaDescription', 'string')
+       */
+      (chainer: 'have.ariaDescription', text: string): Chainable<Subject>;
+      /**
+       * Asserts the element has an aria label matching the text. This
+       * can be from an `aria-label` or an `aria-labelledby`
+       * @example
+       *    cy.get('body').should('have.ariaLabel', 'string')
+       */
+      (chainer: 'have.ariaLabel', text: string): Chainable<Subject>;
+    }
+  }
+}
+
+export const haveAriaDescription = (text: string) => ($target: JQuery) => {
+  expect($target).to.have.attr('aria-describedby');
+
+  const id = $target.attr('aria-describedby');
+  const $descriptionEl = Cypress.$(`[id="${id}"]`);
+  if (!$descriptionEl.length) {
+    throw Error(
+      `Could not find an element with an id matching the aria-describedby: ${$target[0].outerHTML}`
+    );
+  }
+
+  expect($descriptionEl).to.have.text(text);
+};
+
+export const haveAriaLabel = (text: string) => ($target: JQuery) => {
+  if ($target.attr('aria-label')) {
+    expect($target).to.have.attr('aria-label', text);
+  } else if ($target.attr('aria-labelledby')) {
+    const id = $target.attr('aria-labelledby');
+    const $labelledEl = Cypress.$(`[id="${id}"]`);
+    if (!$labelledEl.length) {
+      throw Error(
+        `Could not found an element with an id matching the aria-labelledby:  ${$target[0].outerHTML}`
+      );
+    }
+
+    expect($labelledEl).to.have.text(text);
+  } else {
+    throw Error(`Expected element to have an aria-label or aria-labelledby, but did not find one.`);
+  }
+};
+
+Cypress.Commands.overwrite('should', (originalFn, subject, expectation, ...args) => {
+  const customMatchers = {
+    'have.ariaDescription': haveAriaDescription(args[0]),
+    'have.ariaLabel': haveAriaLabel(args[0]),
+  };
+  // See if the expectation is a string and if it is a member of Jest's expect
+  if (typeof expectation === 'string' && customMatchers[expectation]) {
+    return originalFn(subject, customMatchers[expectation]);
+  }
+
+  return originalFn(subject, expectation, ...args);
```

Comment:
It is possible for these accessibility helpers to be a chai matcher library. I end up using them enough that I decided to add them here.